pepper-apis/src/main/java/org/broadinstitute/ddp/housekeeping/handler/EmailNotificationHandler.java

```diff
@@ -48,242 +43,208 @@
-        if (StringUtils.isNotEmpty(firstName)
-                && StringUtils.isNotEmpty(lastName)) {
-            return String.format("Dear %s %s,", firstName, lastName);
+        // should retry since the send call failed
+        sendResult.rethrowIfThrown(e -> new MessageHandlingException(
+                "Error sending template " + templateId + " to " + distributionList, e, true));
+
+        if (sendResult.getStatusCode() == 200 || sendResult.getStatusCode() == 202) {
+            LOG.info("Sent template {} version {} to {}", templateId, versionUsed, distributionList);
+            new StackdriverMetricsTracker(StackdriverCustomMetric.EMAILS_SENT, studyGuid,
+                    PointsReducerFactory.buildSumReducer()).addPoint(1, Instant.now().toEpochMilli());
         } else {
-            return defaultSalutation;
+            throw new MessageHandlingException("Attempt to send template " + templateId + " to " + distributionList
+                    + " failed with " + sendResult.getStatusCode() + ":" + sendResult.getError(), true);
         }
     }
 
     /**
-     * Checks if a the notification message should be ignored. This can be the case when a user
-     * exited the study or explicitly mentioned that he/she doesn't want to receive notifications
+     * Checks if the notification message should be ignored. This can be the case when a user exited the study or
+     * explicitly mentioned that they do not want to receive notifications.
      *
-     * @param notificationMessage A message to get the user/study information from
-     * @return A boolean value indicating whether the message should be ignored
+     * @param apisHandle      the database handle
+     * @param studyGuid       the study guid
+     * @param participantGuid the participant guid
+     * @return true if message should be ignored, otherwise false
      */
-    private static boolean messageShouldBeIgnoredForParticipant(NotificationMessage notificationMessage) {
-        Optional<EnrollmentStatusType> status = TransactionWrapper.withTxn(
-                TransactionWrapper.DB.APIS,
-                handle -> {
-                    return handle.attach(JdbiUserStudyEnrollment.class).getEnrollmentStatusByUserAndStudyGuids(
-                            notificationMessage.getParticipantGuid(),
-                            notificationMessage.getStudyGuid()
-                    );
-                }
-        );
-        if (status.isPresent() && !status.get().shouldReceiveCommunications()) {
-            LOG.info(
-                    "The participant {} in study {} should not receive communications because of status {}",
-                    notificationMessage.getParticipantGuid(),
-                    notificationMessage.getStudyGuid(),
-                    status.get()
-            );
+    boolean messageShouldBeIgnored(Handle apisHandle, String studyGuid, String participantGuid) {
```

Comment:
I made this method non-static package private so it's easier to test it in isolation.